tests/db_tests/conftest.py

```diff
@@ -178,6 +178,7 @@ def mount(self):
         if not self.mounted:
             print("mounting fake device...")
             self.activity_path_on_device.mkdir(parents=True)
+            os.mknod(f"{self.mount_path}/AUTORUN.INF")
```

Comment:
This line seems to cause the mac pipeline to fail due to a permission error. I would suggest to use [pytest for creating dummy test files](https://docs.pytest.org/en/stable/tmpdir.html) instead of `os.mknod`. Since for the test it only needs to be different to a dir. Or did you chose `mknod` for a specific reason?

Reply:
`os.mknod` was simply a simple solution I found when googling `Python Create file` :laughing:. I did not realise pytest had some own functions to create files etc. I now use the pytest functions and it works on macos now